### PR TITLE
Media group role refacto

### PR DESCRIPTION
**Pull request by Thiblef**
[OO-FEATURE] Media roles gestion for each folders are available in the group panel
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/145
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1416
https://github.com/open-orchestra/open-orchestra-model-interface/pull/163
https://github.com/open-orchestra/open-orchestra/pull/831

### CODE REVIEW

**Comment by Thiblef** [comment]
@itkg-alavieille @itkg-ngilain @itkg-nanne @nicolasThal  PR updated

File: MediaModelBundle/EventListener/UpdateMediaFolderGroupRoleListener.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     *
+     * @throws MediaFolderGroupRoleNotFoundException
+     */
+    public function preUpdate(PreUpdateEventArgs $event)
+    {
+        $document = $event->getDocument();
+        $uow = $event->getDocumentManager()->getUnitOfWork();
+        if (
+            $document instanceof DocumentGroupRoleInterface &&
+            'folder' === $document->getType() &&
```
Comment (itkg-alavieille) [inline MediaModelBundle/EventListener/UpdateMediaFolderGroupRoleListener.php]: use constant

**Comment by Thiblef** [comment]
@itkg-alavieille @itkg-ngilain @itkg-nanne @nicolasThal PR updated

**Comment by itkg-alavieille** [comment]
:+1:  when test run

**Comment by itkg-ngilain** [comment]
:+1:

**Comment by Thiblef** [comment]
@itkg-alavieille @itkg-ngilain @itkg-nanne @nicolasThal PR updated

**Comment by itkg-alavieille** [comment]
:+1:

File: Media/Model/FolderInterface.php [hunk 2]
```diff
@@ -11,6 +11,9 @@
  */
 interface FolderInterface extends TimestampableInterface, BlameableInterface
 {
+    const GROUP_ROLE_TYPE = 'folder';
+    const ROOT_PARENT_ID = '-';
```
Comment (itkg-ngilain) [inline Media/Model/FolderInterface.php]: ROOT_PARENT_ID or simply ROOT_ID ?

File: MediaModelBundle/Repository/FolderRepository.php [hunk 3]
```diff
@@ -38,4 +38,38 @@ public function findAllRootFolderBySiteId($siteId)
...
+
+    /**
+     * @param string      $siteId
+     * @param string|null $parentId
+     *
+     * @throws \Exception
+     *
+     * @return Collection
+     */
+    public function findBySiteId($siteId, $parentId = null)
```
Comment (itkg-ngilain) [inline MediaModelBundle/Repository/FolderRepository.php]: findBySiteId or simply findBySite ?

File: Media/Model/FolderInterface.php (hunk 2 above)
Comment (Thiblef) [inline Media/Model/FolderInterface.php]: It's `ROOT_PARENT_ID`, like for node, the root node parent id is '-'.

File: MediaModelBundle/Repository/FolderRepository.php (hunk 3 above)
Comment (Thiblef) [inline MediaModelBundle/Repository/FolderRepository.php]: `findBySiteId`, the param of the method is a siteId string.

**Pull request merged**